### PR TITLE
Revised the implemented of the FlowDurationCurveDeviationRecorder

### PR DESCRIPTION
**Pull request by jetuk**
The recorder now supports an upper and lower target FDC. Deviation is calculated as positive above
and negative below the respective FDC. If flow is between the two zero deviation is returned.

@Batch21 could give this a read please.

### CODE REVIEW

**Comment by Batch21** [comment]
It looks good. I know it's not needed currently, but would it be worth recording the direction of the deviation?

**Comment by jetuk** [comment]
That's a good point. It also doesn't have a mechanism to find which percentile has the worst deviation.

**Comment by jetuk** [comment]
I think this does what we want now.

File: pywr/recorders/_recorders.pyx [hunk 1]
```diff
@@ -442,66 +442,90 @@ FlowDurationCurveRecorder.register()
...
+                raise ValueError("The number of upper target FDCs does not match the number of scenarios")
 
     cpdef finish(self):
         super(FlowDurationCurveDeviationRecorder, self).finish()
 
         cdef int i, j, k, sc_index
         cdef ScenarioIndex scenario_index
-        cdef double[:] trgt_fdc
+        cdef double[:] utrgt_fdc, ltrgt_fdc
+        cdef udev, ldev
```
Comment (snorfalorpagus) [inline pywr/recorders/_recorders.pyx]: These should be typed double.

**Pull request merged**